thirdparty/DiskANN/src/aux_utils.cpp

```diff
@@ -745,9 +746,7 @@ namespace diskann {
       }));
     }
 
-    for (auto &future : futures) {
-      future.wait();
-    }
+    knowhere::WaitAllSuccess(futures);
```

Comment:
memory leaks in case of exception
```
    if (samples != nullptr)
      delete[] samples;
    if (pq_code != nullptr)
      delete[] pq_code;
```